addons/utm/models/utm.py

```diff
@@ -42,14 +49,84 @@ def _group_expand_stage_ids(self, stages, domain, order):
         stage_ids = stages._search([], order=order, access_rights_uid=SUPERUSER_ID)
         return stages.browse(stage_ids)
 
+    def _merge_utm_campaigns(self, campaign=False, name=False, user_id=False, stage_id=False, tag_ids=False):
+        if len(self.ids) <= 1:
+            raise UserError('Please select more than one campaign from the list.')
+        if not all(r.active for r in self):
+            raise UserError('Only active campaigns can be merged.')
+        if campaign and campaign not in self:
+            raise UserError('Kept campaign should be one of the merged campaign.')
+
+        merged_campaign = campaign or self[0]
+
+        merged_data = self._merge_data()
+        if name:
+            merged_data['name'] = name
+        if user_id:
+            merged_data['user_id'] = user_id
+        if stage_id:
+            merged_data['stage_id'] = stage_id
+        if tag_ids:
+            merged_data['tag_ids'] = tag_ids
+        merged_campaign.write(merged_data)
+
+        deactived_campaigns = self - merged_campaign
+        deactived_campaigns.write({'active': False, 'reference_utm_campaign_id': merged_campaign.id})
```

Comment:
Based on what we just discussed, I think you should check here that the campaings you're de-activating are not referenced by other campaigns and if so, update their link.

So something like:

```
self.search([('reference_utm_campaign_id', 'in', deactived_campaigns.ids)].write({
    'reference_utm_campaign_id': merged_campaign.id
})
```

So everything is cleaner and you don't have to chain through the deactivated ones when you update your cookies.